qiskit_nature/algorithms/initial_points/hf_initial_point.py

```diff
@@ -0,0 +1,203 @@
+    def grouped_property(self, grouped_property: GroupedSecondQuantizedProperty) -> None:
+        if not isinstance(grouped_property, GroupedSecondQuantizedProperty):
+            logger.warning(
+                "grouped_property is not of type GroupedSecondQuantizedProperty; "
+                "grouped_property and reference_energy will not be set."
+            )
+            return
+
+        electronic_energy: ElectronicEnergy | None = grouped_property.get_property(ElectronicEnergy)
+        if not isinstance(electronic_energy, ElectronicEnergy):
+            logger.warning(
+                "ElectronicEnergy not obtained from grouped_property; "
+                "grouped_property and reference_energy will not be set."
+            )
+            return
+
+        self._reference_energy = electronic_energy.reference_energy if not None else 0.0
+        self._grouped_property = grouped_property
+
+    @property
+    def ansatz(self) -> UCC:
+        """The UCC ansatz.
+
+        This is used to ensure that the :attr:`excitation_list` matches with the UCC ansatz that
+        will be used with the VQE algorithm.
+
+        Raises:
+            QiskitNatureError: If not set using a valid
+                :class:`~qiskit_nature.circuit.library.ansatzes.ucc.UCC` instance.
+        """
+        return self._ansatz
+
+    @ansatz.setter
+    def ansatz(self, ansatz: UCC) -> None:
+        if not isinstance(ansatz, UCC):
+            raise QiskitNatureError(f"Expected a UCC ansatz, but got type: {type(ansatz)}.")
+
+        # Operators must be built early to compute the excitation list.
+        _ = ansatz.operators
+
+        # Invalidate any previous computation.
+        self._corrections = None
+
+        self._excitation_list = ansatz.excitation_list
+        self._ansatz = ansatz
+
+    @property
+    def excitation_list(self) -> list[tuple[tuple[int, ...], tuple[int, ...]]]:
+        """The list of excitations.
+
+        Setting this will overwrite the excitation list from the ansatz.
+        """
+        return self._excitation_list
+
+    @excitation_list.setter
+    def excitation_list(self, excitations: list[tuple[tuple[int, ...], tuple[int, ...]]]):
+        # Invalidate any previous computation.
+        self._corrections = None
+
+        self._excitation_list = excitations
+
+    def to_numpy_array(self) -> np.ndarray:
+        """The initial point as an array."""
+        if self._corrections is None:
+            self.compute()
+        return np.asarray([value[0] for value in self._corrections.values()])
+
+    def get_energy_corrections(self) -> np.ndarray:
+        """The energy corrections for each excitation."""
+        if self._corrections is None:
+            self.compute()
+        return np.asarray([value[1] for value in self._corrections.values()])
+
+    def get_energy_correction(self) -> float:
+        """The overall energy correction."""
+        return self.get_energy_corrections().sum()
```

Comment:
I was a bit confused to see all this logic here because all these energy terms are simply zero in the HF case, but I see that you made the MP2 point a subclass of this.
I am wondering whether it would be better to keep the HF case as simple as possible and instead have this logic only on the MP2 case. But presumably the HF point will still need these methods in place (but return zero) for API consistency, right?

What does this look like compared to VSCF? Could it make sense to extract base implementations of these methods into the `InitialPoint` super class?

Reply:
To clarify: I am not suggesting that the generated initial point should be an all-zero vector by default. But instead, the returned energy correction could default to being `0.0`.

Reply:
For HF and VSCF one could obviously just return a NumPy all-zero array of the appropriate size. This would make their logic way more transparent. I kept the dictionary simply to keep the logic in MP2 the same, but given that HFInitialPoint is likely to be used far more often than MP2, I'm tempted to dump the dictionary... yeah, I've talked myself into doing that.